### PR TITLE
[kubernetes] Add support for Ubuntu

### PR DESCRIPTION
**Pull request by slashdd**
There are multiples ways to run a Kubernetes cluster with Ubuntu.
This commit is based on the "Official Ubuntu Guides", The Canonical
Distribution of Kubernetes deployed using "conjure-up".

Signed-off-by: Eric Desrochers <eric.desrochers@canonical.com>

---
Please place an 'X' inside each '[]' to confirm you adhere to our [Contributor Guidelines](https://github.com/sosreport/sos/wiki/Contribution-Guidelines)

- [x] Is the commit message split over multiple lines and hard-wrapped at 72 characters?
- [x] Is the subject and message clear and concise?
- [x] Does the subject start with **[plugin_name]** if submitting a plugin patch or a **[section_name]** if part of the core sosreport code?
- [x] Does the commit contain a **Signed-off-by: First Lastname <email@example.com>**?

### CODE REVIEW

**Comment by slashdd** [comment]
Related documents:
https://kubernetes.io/docs/getting-started-guides/ubuntu/#official-ubuntu-guides
https://www.ubuntu.com/kubernetes

File: sos/plugins/kubernetes.py [hunk 1]
```diff
@@ -9,13 +9,29 @@
...
-    """Kubernetes plugin
+    """Ubuntu Kubernetes plugin
+    """
+
+    files = ("/snap/bin/kubectl",)
+
+    def setup(self):
+        self.add_cmd_output([
+            "snap run kubectl get nodes -o wide",
+            "snap run kubectl cluster-info",
```
Comment (TurboTurtle) [inline sos/plugins/kubernetes.py]: This is really the only new command being captured by `Ubuntukubernetes`, the other bits are all already captured. Is there any reason we couldn't add `cluster-info` to the existing collection and just set `kube_cmd` to `snap run kubectl` for Ubuntu while keeping it `kubectl` elsewhere?

Comment (slashdd) [reply sos/plugins/kubernetes.py]: Sure, I'll have a look at it and re-submit a PR later.

File: sos/plugins/kubernetes.py [hunk 2]
```diff
@@ -35,7 +35,22 @@ class kubernetes(Plugin, RedHatPlugin):
...
         if any([
             path.exists("/var/run/kubernetes/apiserver.key"),
-            path.exists("/etc/origin/master/master-config.yaml")
+            path.exists("/etc/origin/master/master-config.yaml"),
+            path.exists("/root/cdk/server.key")
+        ]):
+            return True
+        return False
+
+    def check_is_ubuntu(self):
```
Comment (TurboTurtle) [inline sos/plugins/kubernetes.py]: The Plugin class already has this kind of OS-detection functionality built-in with the tagging mechanic.

This means we can just move `kube_cmd` to a class attr, and then use something like this for Ubuntu:

```python
class UbuntuKubernetes(kubernetes, UbuntuPlugin):

    files = ("/root/cdk/server.key", )

    def setup():
        self.kube_cmd = "snap run kubectl "
        super(UbuntuKubernetes, self).setup()
```

After that we'd just need to either transform `self.kube_cmd` back into `kube_cmd` in `Kubernetes.setup()` or modify all references of kube_cmd to be `self.kube_cmd` (probably best).

**Comment by bmr-cymru** [comment]
No response to review comments since last July: if this functionality is still required, please re-open this pull request.

**Pull request closed without merging**